appservice/src/createAppService/SiteNameStep.ts

```diff
@@ -113,6 +113,8 @@ export class SiteNameStep extends AzureNameStep<SiteNameStepWizardContext> {
 
         if (name.length < siteNamingRules.minLength || name.length > siteNamingRules.maxLength) {
             return vscode.l10n.t('The name must be between {0} and {1} characters.', siteNamingRules.minLength, siteNamingRules.maxLength);
+        } else if (this._siteFor === "containerizedFunctionApp" && (!/^[a-z]([-a-z0-9]*[a-z0-9])?$/.test(name))) {
```

Comment:
I think the `*` might be switched up here, otherwise I think you can still have multiple `-`'s.

There's a pretty sweet regexp you made for `ContainerAppNameStep` that might be good to copy over here.  Seems like the rules might be pretty similar

Reply:
Oh lol yes I copied this from `ManagedEnvironmentNameStep` since I figured it was the same between the two but I can change it 

On another note shouldn't the regex be the same between `ContainerAppNameStep` and `ManagedEnvironmentNameStep`  since the error message is the same between the two?

Reply:
Yeah good catch, if the requirements are the same I think we should update that check logic